filestore/pb/dataobj.proto

```diff
@@ -4,4 +4,5 @@ message DataObj {
         optional string FilePath = 1;
         optional uint64 Offset = 2;
         optional uint64 Size = 3;
+        optional bool URL = 4;
```

Comment:
I'd make this an enum

Reply:
Agreed. Making this a type enum makes more sense and allows for better future extensibility.

Reply:
Alternatively, we could:

1. Rename `FilePath` to `Location`.
2. Use URLs for everything and use the `file:///` schema for files.
3. Unify the commands.

This should make this trivial to extend via plugins.

(don't bother if this leads to a bunch of additional work)

Reply:
> (2) Use URLs for everything and use the file:/// schema for files.

This will require a migration.  Alternatively if I am correct that paths or stored absolute, we can use that to distinguish between url's and files.

> (3)  Unify the commands.

This makes sense since they are essentially doing the same thing, and I don't think it will be too hard.

Reply:
> This will require a migration. Alternatively if I am correct that paths or stored absolute, we can use that to distinguish between url's and files.

I assume they're absolute. And yes, that'd be a reasonable way forward.

Reply:
Actually there not.  However the string "http://` or `https://` will not appear in any normalized file path as `//` will always be simplified to '/'.  So I still think checking for a `http://` or `https://` is valid way forward and can make this patch fairly small and non-invasive.